text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/parser/TokenParser.java

```diff
@@ -503,6 +503,8 @@ private static ElementNode buildTree(
       throw new ParsingException(sb.toString(), message, errorTokens);
     }
 
+    tryToNag(message, strict);
```

Comment:
Why are you running this so late? Plus, buildTree isn't used for the strip and escape tokens methods.